src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private final int radius;
+
+    public Circle(int radius) {
+        this.radius = radius;
+        setColor(new ColorSupplier().getRandomColor());

```

Comment:
Please use ColorSupplier in FigureSupplier to get color and pass this value into constructor.
Comment valid for all figures!

Reply:
if you already approach exactly like you did with the last comment, then I don’t think that you are right about this comment, since the condition says - "For this purpose create two more classes:

ColorSupplier with public String getRandomColor() method - for generating random color,
and FigureSupplier with public Figure getRandomFigure() method - for generating figure with random properties."

Reply:
I agree with Uliana, we should separate concerns, Circle shouldn't be concerned with what is the business logic/rules for color setting, it should only know what color is set to it - this will make the class more flexible and general-purpose. If the logic for generating color changed, you would just need to make changes to FigureSupplier, instead of going thru all the figures and rewriting constructors